picard/util/checkupdate.py

```diff
@@ -0,0 +1,160 @@
+                1 = stable and beta releases are checked.
+                2 = stable, beta and dev releases are checked.
+
+        Returns:
+            none.
+
+        Raises:
+            none.
+        '''
+        self._show_always = show_always
+        self._update_level = update_level
+
+        if self._available_versions['stable']['tag']:
+            # Release information already acquired from specified website api.
+            self._display_results()
+        else:
+            # Gets list of releases from specified website api.
+            self._query_available_updates(callback=callback)
+
+    @property
+    def available_versions(self):
+        '''Provide a list of the latest version tuples for each update type.'''
+        return self._available_versions
+
+    def _query_available_updates(self, callback=None):
+        '''Gets list of releases from specified website api.'''
+        output_text = _("Getting release information from %s." % (VERSIONS_API['host'],))
+        log.debug(output_text)
+        self.tagger.webservice.get(
+            VERSIONS_API['host'],
+            VERSIONS_API['port'],
+            VERSIONS_API['endpoint'],
+            partial(self._releases_json_loaded, callback=callback),
+            parse_response_type=None,
+            priority=True,
+            important=True
+        )
+
+    def _releases_json_loaded(self, response, reply, error, callback=None):
+        '''Processes response from specified website api query.'''
+        if error:
+            log.error(
+                N_("Error loading releases list: %(error)s"),
+                {'error': reply.errorString()},
+            )
+            if self._show_always:
+                msg_title = _("Picard Update")
+                msg_text = _("Unable to retrieve the latest version information.")
+                QMessageBox.information(
+                    self._parent, msg_title, msg_text, QMessageBox.Ok, QMessageBox.Ok)
+        else:
+            self._available_versions = load_json(response)['versions']
+            for key in self._available_versions.keys():
+                log.debug("Version key '%s' --> %s" %
+                          (key, self._available_versions[key],))
+            self._display_results()
+        if callback:
+            callback(not error)
+
+    def _display_results(self):
+        # Display results to user.
+        msg_title = _("Picard Update")
+        key = ''
+        high_version = PICARD_VERSION
+        i = 0
+        for test_key in ['stable', 'beta', 'dev']:
+            if self._update_level >= i and  compare_version_tuples(high_version, self._available_versions[test_key]['version']) > 0:
+                key = test_key
+                high_version = self._available_versions[test_key]['version']
+            i += 1
+        if key:
+            msg_text = _("A new version of Picard is available.\n\nOld version: %s\nNew version: %s\n\n"
```

Comment:
Keep title and message within `QMessageBox.information()` it helps translators to have the context.
Also, now we moved to Python3, use placeholders/format() and wrap lines according to embedded newlines:

```python
if QMessageBox.information(
    self._parent,
    _("Picard Update"),
    _("A new version of Picard is available.\n\n"
      "Old version: {picard_old_version}\n"
      "New version: {picard_new_version}\n\n"
      "Would you like to download the new version?").format(
         picard_old_version=PICARD_FANCY_VERSION_STR,
         picard_new_version=self._available_versions[key]['tag']
       )
) == QMessageBox.Ok:
    wb2.open(self._available_versions[key]['urls']['download'])
```

Reply:
Reformatted as requested.